src/app/features/collectibles/ordinals.tsx

```diff
@@ -1,89 +1,91 @@
 
-interface OrdinalsProps {
-  query: ReturnType<typeof useGetOrdinalsQuery>;
+interface InscriptionProps {
+  path: string;
 }
-export function Ordinals({ query }: OrdinalsProps) {
-  const { isLoading, isError, data } = query;
 
-  // const qMockImages = useQuery(['mock'], async () => {
-  //   const res = await fetch('https://picsum.photos/v2/list');
-  //   const resParsed = await res.json();
-  //   return resParsed.map(i => ({
-  //     title: 'Foo title',
-  //     content: i.download_url,
-  //     preview: `http://localhost:1234?dl=${i.download_url}`,
-  //   }));
-  // });
-  // if (Math.random() < 1) {
-  //   return (
-  //     <>
-  //       <p>working?</p>
-  //       {qMockImages.data &&
-  //         qMockImages.data
-  //           .slice(12)
-  //           .map(ordinal => (
-  //             <ImageCollectible
-  //               onClick={() => alert('clicked')}
-  //               src={ordinal.content}
-  //               subtitle="Ordinal inscription"
-  //               title={ordinal.title}
-  //             />
-  //           ))}
-  //     </>
-  //   );
-  // }
+function Inscription({ path: id }: InscriptionProps) {
+  const { isLoading, isError, data } = useInscriptionQuery(id);
+
+  if (isLoading) return null; // TODO
+
+  if (isError) return null; // TODO
+
+  const ordinal = whenOrdinalType(data['content type'], {
+    image: () => ({
+      type: 'image',
+      title: data.title,
+      infoUrl: createInfoUrl(data.content),
+      content: `https://ordinals.com${data.content}`,
+    }),
+    other: () => ({
+      type: 'other',
+      title: data.title,
+      infoUrl: `https://ordinals.com${data.content}`.replace('content', 'inscription'),
+    }),
+  });
 
-  // if (Math.random() < 1) {
-  //   return (
-  //     <OtherCollectible
-  //       onClick={() => alert('clicked')}
-  //       subtitle="Ordinal inscription"
-  //       title="Foo"
-  //     />
-  //   );
-  // }
+  switch (ordinal.type) {
+    case 'image': {
+      return (
+        <ImageCollectible
+          key={ordinal.infoUrl}
+          onClick={() => openInNewTab(ordinal.infoUrl)}
+          src={ordinal.content}
+          subtitle="Ordinal inscription"
+          title={ordinal.title}
+        />
+      );
+    }
+    case 'other': {
+      return (
+        <OtherCollectible
+          key={ordinal.infoUrl}
+          onClick={() => openInNewTab(ordinal.infoUrl)}
+          subtitle="Ordinal inscription"
+          title={ordinal.title}
+        />
+      );
+    }
+    default: {
+      return null;
+    }
+  }
+}
+
+interface TransactionQueryLoaderProps {
+  q: UseQueryResult<OrdApiXyzGetTransactionOutput>;
+}
+function TransactionQueryLoader({ q }: TransactionQueryLoaderProps) {
+  if (q.isLoading) return null; // TODO
 
-  if (isLoading) return null;
+  if (q.isError) return null; // TODO
 
-  // TODO: user facing error handling.
-  if (isError) return null;
+  return <Inscription path={q.data.inscriptions} />;
+}
```

Comment:
This pattern differs from how it's done on homepage. If we were to follow that style, here we'd return the inscription data to `children()`, a fn, so that below it's consumed as:


```jsx
<TransactionQueryLoader key={i} q={q}>
  {inscriptions => <Inscription path={inscriptions} />}
</<TransactionQueryLoader>
```

This would be a more compositional approach, and is more readable when viewing the feature (you don't need to dive into component files) to see what's there. It also scopes the functionality of `TransactionQueryLoader` to loading the data, where as right now, it both loads the data, and renders the child. 

Imo I think that's a cleaner abstractions/division of responsibilities, but if it were to be left as is, I wouldn't call it a "Loader" component, because it does more than that. It's a general purpose `<Inscriptions />` feature.

Reply:
Refactored to use `children` as inline function